source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,371 @@
+      stats.counter(counter_name_lookup[cluster_name]["upstream_rq_4xx"]).value() +
+      stats.counter(counter_name_lookup[cluster_name]["retry.upstream_rq_4xx"]).value() -
+      cluster_stats.upstream_rq_timeout_.value();
+
+  pushNewValue(counter_name_lookup[cluster_name]["errors"], errors);
+
+  uint64_t success = stats.counter(counter_name_lookup[cluster_name]["upstream_rq_2xx"]).value();
+  pushNewValue(counter_name_lookup[cluster_name]["success"], success);
+
+  uint64_t rejected = cluster_stats.upstream_rq_pending_overflow_.value();
+  pushNewValue(counter_name_lookup[cluster_name]["rejected"], rejected);
+
+  // should not take from upstream_rq_total since it is updated before its components,
+  // leading to wrong results such as error percentage higher than 100%
+  uint64_t total = errors + timeouts + success + rejected;
+  pushNewValue(counter_name_lookup[cluster_name]["total"], total);
+
+  ENVOY_LOG(trace, "{}", printRollingWindow());
+}
+
+void HystrixStatCache::resetRollingWindow() { rolling_stats_map_.clear(); }
+
+void HystrixStatCache::addStringToStream(absl::string_view key, absl::string_view value,
+                                         std::stringstream& info) {
+  std::string quoted_value = absl::StrCat("\"", value, "\"");
+  addInfoToStream(key, quoted_value, info);
+}
+
+void HystrixStatCache::addIntToStream(absl::string_view key, uint64_t value,
+                                      std::stringstream& info) {
+  addInfoToStream(key, std::to_string(value), info);
+}
+
+void HystrixStatCache::addInfoToStream(absl::string_view key, absl::string_view value,
+                                       std::stringstream& info) {
+  if (!info.str().empty()) {
+    info << ", ";
+  }
+  std::string added_info = absl::StrCat("\"", key, "\": ", value);
+  info << added_info;
+}
+
+void HystrixStatCache::addHystrixCommand(std::stringstream& ss, absl::string_view cluster_name,
+                                         uint64_t max_concurrent_requests, uint64_t reporting_hosts,
+                                         uint64_t rolling_window) {
+  std::stringstream cluster_info;
+  std::time_t currentTime = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+  addStringToStream("type", "HystrixCommand", cluster_info);
+  addStringToStream("name", cluster_name, cluster_info);
+  addStringToStream("group", "NA", cluster_info);
+  addIntToStream("currentTime", static_cast<uint64_t>(currentTime), cluster_info);
+  addInfoToStream("isCircuitBreakerOpen", "false", cluster_info);
+
+  uint64_t errors = getRollingValue(cluster_name, "errors");
+  uint64_t timeouts = getRollingValue(cluster_name, "timeouts");
+  uint64_t rejected = getRollingValue(cluster_name, "rejected");
+  uint64_t total = getRollingValue(cluster_name, "total");
+
+  uint64_t error_rate =
+      total == 0
+          ? 0
+          : (static_cast<double>(errors + timeouts + rejected) / static_cast<double>(total)) * 100;
```

Comment:
nit: I think you can remove all the static casts by doing the multiply on the numerator before dividing:
```
(100 * (errors + timeouts + rejected)) / total;
```